src/6_branch.js

```diff
@@ -1061,8 +1061,14 @@ Branch.prototype['logEvent'] = wrap(callback_params.CALLBACK_ERR, function(done,
 /*** +TOC_HEADING &Deep Linking& ^ALL ***/
 /*** +TOC_ITEM #linkdata-callback &.link()& ^ALL ***/
 Branch.prototype['link'] = wrap(callback_params.CALLBACK_ERR_DATA, function(done, data) {
+	var dataCopy = utils.merge({}, data);
+	var keyCopy = this.branch_key;
 	this._api(resources.link, utils.cleanLinkData(data), function(err, data) {
```

Comment:
Since you will need the result of `utils.cleanLinkData(data)` here and in `generateDynamicBNCLink`, why don't you just pull this out into a variable above instead of making a copy of data.